src/main/webapp/app.js

```diff
@@ -80,35 +80,50 @@ var signInWithPopup = function() {
  * @param {!firebase.User} user
  */
 var handleSignedInUser = function(user) {
-  document.getElementById('user-signed-in').style.display = 'block';
-  document.getElementById('user-signed-out').style.display = 'none';
-  document.getElementById('name').textContent = user.displayName;
-  document.getElementById('email').textContent = user.email;
-  document.getElementById('phone').textContent = user.phoneNumber;
-  if (user.photoURL) {
-    var photoURL = user.photoURL;
-    // Append size to the photo URL for Google hosted images to avoid requesting
-    // the image with its original resolution (using more bandwidth than needed)
-    // when it is going to be presented in smaller size.
-    if ((photoURL.indexOf('googleusercontent.com') != -1) ||
-        (photoURL.indexOf('ggpht.com') != -1)) {
-      photoURL = photoURL + '?sz=' +
-          document.getElementById('photo').clientHeight;
+  // This function should only be reached if the user is signed in.
+  if (window.location.pathname !== '/loggedIn.html') {
+    // User will be redirected if he is not on the right page.
+    window.location.pathname = '/loggedIn.html';
+  }
+  if (document.getElementById('user-signed-in')) {
```

Comment:
This if was also added because data doesn't load fast enough? If that is the case, don't we have to wait for it to be loaded and then make the changes? I'm asking just to be sure I understood correctly.

Reply:
No, that was added because I considered the user that is signed up should only see the interface on loggedIn.html and a user that is not signed should only see the index page where he is asked to sign in.

Reply:
Basically we need to check that the user is not accessing Pages that he is not allowed to see by changing the URL by himselft.